sort.js

```diff
@@ -8,19 +9,48 @@ node sort.js {arguments}
 
 The script updates each file by sorting the labels by their keys.
 
+If no arguments are passed, all language files will be sorted.
+
 Example:
 node sort.js languages/de-DE/assess-api.json languages/de-DE/questions-api.json
+node sort.js
 */
-args.forEach((arg) => {
-    const text = fs.readFileSync(arg);
-    const parsedJson = JSON.parse(text);
-    const sortedKeys = Object.keys(parsedJson).sort();
-    let result = {};
-    
-    sortedKeys.forEach((key) => {
-        result[key] = parsedJson[key];
-    });          
+let files;
+
+if (args.length) {
+    files = args;
+} else {
+    files = execSync("find ./languages -type f -name '*.json'", {
+        encoding: 'utf8'
+    }).trim().split(/[\r\n]+/gm);
+}
+
+files.forEach((file) => {
+    const text = fs.readFileSync(file);
+    let result = JSON.parse(text);
     
+    result = recursiveSort(result);
     result = JSON.stringify(result, null,'    ');
-    fs.writeFileSync(arg, result); 
+    fs.writeFileSync(file, result); 
 });
+
+function recursiveSort(obj) {
+    const sortedKeys = Object.keys(obj).sort();
+    let result = {};
+    
+    sortedKeys.forEach((key) => {
+        const val = obj[key];
+
+        // Do not sort arrays, as the order of elements is important.
+        // E.g. reordering questions-api.json's characterMapCharacters would
+        // cause the special characters to appear in a different order in
+        // the product.
+        if (val && !Array.isArray(val) && typeof val === 'object') {
```

Comment:
Do you need the `!Array.isArray(val)` here? Or are arrays `object`s in JS? If so, I'd swap the `Array.isArray` and the `typeof`

Reply:
@shtrom welcome to `typeof` confusion land :)
![image](https://user-images.githubusercontent.com/235000/69769331-87f7e600-11d8-11ea-86cd-4b0f48ff7bf6.png)

http://web.mit.edu/jwalden/www/isArray.html

Reply:
Yep, arrays are considered `object`s in JS. I'll swap the two then.